File: android/src/oldarch/com/appodeal/rnappodeal/RNAppodealMrecViewManager.java
package com.appodeal.rnappodeal;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;

import java.util.Map;

public class RNAppodealMrecViewManager extends SimpleViewManager<RCTAppodealMrecView> {

    private final RNAppodealMrecViewManagerImpl mrecViewManagerImpl;

    public RNAppodealMrecViewManager() {
        this.mrecViewManagerImpl = new RNAppodealMrecViewManagerImpl();
    }

    @NonNull
    @Override
    public String getName() {
        return RNAppodealMrecViewManagerImpl.NAME;
    }

    @NonNull
    @Override
    protected RCTAppodealMrecView createViewInstance(@NonNull ThemedReactContext reactContext) {
        return mrecViewManagerImpl.createViewInstance(reactContext);
    }

    @Override
    public void onDropViewInstance(@NonNull RCTAppodealMrecView view) {
        super.onDropViewInstance(view);
        mrecViewManagerImpl.onDropViewInstance(view);
    }

    @Nullable
    @Override
    public Map<String, Object> getExportedViewConstants() {
        return mrecViewManagerImpl.getExportedViewConstants();
    }

    @ReactProp(name = "placement")
    public void setPlacement(RCTAppodealMrecView view, @Nullable String value) {
        if (value != null) {
            view.setPlacement(value);
        }
    }
} 

File: android/src/newarch/com/appodeal/rnappodeal/RNAppodealMrecViewManager.kt
package com.appodeal.rnappodeal

import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.ViewManagerDelegate
import com.facebook.react.viewmanagers.RNAppodealMrecViewManagerDelegate
import com.facebook.react.viewmanagers.RNAppodealMrecViewManagerInterface

@ReactModule(name = RNAppodealMrecViewManagerImpl.NAME)
class RNAppodealMrecViewManager() :
    SimpleViewManager<RCTAppodealMrecView>(),
    RNAppodealMrecViewManagerInterface<RCTAppodealMrecView> {

    private val mrecViewManagerDelegate: ViewManagerDelegate<RCTAppodealMrecView> =
        RNAppodealMrecViewManagerDelegate<RCTAppodealMrecView, RNAppodealMrecViewManager>(this)

    private val mrecViewManagerImpl = RNAppodealMrecViewManagerImpl()

    override fun getDelegate(): ViewManagerDelegate<RCTAppodealMrecView> =
        mrecViewManagerDelegate

    override fun getName(): String = RNAppodealMrecViewManagerImpl.NAME

    override fun createViewInstance(reactContext: ThemedReactContext): RCTAppodealMrecView {
        return mrecViewManagerImpl.createViewInstance(reactContext)
    }

    override fun onDropViewInstance(view: RCTAppodealMrecView) {
        super.onDropViewInstance(view)
        return mrecViewManagerImpl.onDropViewInstance(view)
    }

    override fun getExportedViewConstants(): Map<String?, Any?>? {
        return mrecViewManagerImpl.getExportedViewConstants()
    }

    override fun setPlacement(view: RCTAppodealMrecView, value: String?) {
        value?.let { view.placement = it }
    }
}